lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -344,7 +344,7 @@ private void validatePlatform(DOMDocument domDocument, List<Diagnostic> list, DO
                     list.add(new Diagnostic(range, message, DiagnosticSeverity.Error, LIBERTY_LEMMINX_SOURCE));
                 }
                 preferredPlatformsWithoutVersion.add(platformNoVersionLower);
-                preferredPlatforms.add(platformNameLowerCase);
+                preferredPlatforms.add(platformName);
```

Comment:
There's a reason we store them in lowercase. Liberty let's the user enter feature names and I presume platform names in any case they want. We should do all comparisons/matching based on lower case names. It's the only way we can ensure proper functionality.

Reply:
got it.. changed in the logic where we fetch the platforms from json. changed to lowercase